qiskit/transpiler/passes/routing/stochastic_swap.py

```diff
@@ -24,12 +24,7 @@
 from qiskit.circuit.library.standard_gates import SwapGate
 from qiskit.transpiler.layout import Layout
 
-# pylint: disable=no-name-in-module
-from .cython.stochastic_swap.utils import nlayout_from_layout
-
-# pylint: disable=no-name-in-module
-from .cython.stochastic_swap.swap_trial import swap_trial
-
+from qiskit import stochastic_swap_rs
```

Comment:
Well my bigger concern was more around packaging here longer term. Unless we create rust workspaces (basically submodules with separate `Cargo.toml` files, see [`retworkx-core`](https://github.com/Qiskit/retworkx/tree/main/retworkx-core) for an example of this) the rust project basically ends up as a single binary with a single compiled module. We can add submodules pretty easily to that, but I don't know of an easy way, besides separate workspaces (which seems heavyweight here), to create multiple binary modules from a single rust project.

Basically I have another WIP branch locally on top of this which ports the other cython module for the pauli expectation value calculation to rust (per your earlier point about standardizing on a single compiled language, which is a sentiment others also shared in this week's meeting). But if we did that we'd need a common shared module for all the compiled code. In that local branch I renamed this to `qiskit_core_routines` (I didn't think about the `_` prefix which is a good idea), so `qiskit.qiskit_core_routines.swap_trials` or `qiskit.qiskit_core_routines.density_expval_pauli_with_x`. Although I think maybe adding submodules makes sense so maybe a better end state would be `qiskit._core_routines.stochastic_swap.swap_trials` and `qiskit._core_routines.pauli_exp_val.density_expval_pauli_with_x`

Reply:
Yeah, I thought the response might be something along those lines.  I'm entirely happy to have it in a protected submodule of `qiskit`, like `qiskit._accelerate` or `qiskit._core_routines`.  Submodules with Rust are a good idea, I think, because then from Python you'll be able to do things like
```python
from qiskit._core_routines import stochastic_swap
```
and access things with dotted access to give better context.

Reply:
I'll pivot the PR to do this now, it's simple enough

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7658/commits/7af1c4331d9f26c27b9ad5f101ff5a8c6bc02bac

Reply:
I noticed this too but thought it was just part of the "WIP" designation. For a directory if it's inefficient to put this under e.g. the 1st level directories (tanspiler, dagcircuit, etc) then `_accelerate` or `_extensions` sounds good. `core_routines` might be confusing if we support  python version of the same "core routines".

Reply:
Yeah it definitely was part of the WIP. :) In https://github.com/Qiskit/qiskit-terra/commit/7af1c4331d9f26c27b9ad5f101ff5a8c6bc02bac I updated it to be `_accelerate`. So to get to `swap_trials()` the full path is `qiskit._accelerate.stochastic_swap.swap_trials`.